packages/py-moose-lib/moose_lib/dmv2/ingest_pipeline.py

```
@@ -160,6 +161,14 @@ def __init__(self, name: str, config: IngestPipelineConfig, **kwargs):
                 if isinstance(config.table, OlapConfig)
                 else OlapConfig(life_cycle=config.life_cycle)
             )
+
+            # Validate that the engine is not read-only (Merge engine cannot be written to)
+            if isinstance(table_config.engine, MergeEngine):
+                raise ValueError(
+                    f'IngestPipeline "{name}": Cannot use Merge engine as a table destination. '
+                    f"The Merge engine is read-only and does not support INSERT operations."
+                )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)